packages/shelter/src/util.ts

```diff
@@ -86,6 +86,65 @@ export function createSubscription<TState, TStoreData = any>(
   return data;
 }
 
+
+const storeInitPromises = new WeakMap<FluxStore, Promise<void>>();
+
+// awaits until the _isInitialized property of a store is true by overwritting it's setter
+function awaitStoreInit(store: FluxStore): Promise<void> {
+  return storeInitPromises[store] ?? (storeInitPromises[store] = new Promise<void>(resolve => {
```

Comment:
WeakMap uses `get` and `set` methods, not index syntax.

I'd not expect this to work, as the stores would be coerced to `"[object Object]"` and stored on ony one key.